pkg/image/registry/image/strategy_test.go

```diff
@@ -0,0 +1,106 @@
+package image
+
+import (
+	"math/rand"
+	"reflect"
+	"testing"
+
+	"github.com/google/gofuzz"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	apitesting "k8s.io/kubernetes/pkg/api/testing"
+	"k8s.io/kubernetes/pkg/util/diff"
+
+	"github.com/openshift/origin/pkg/api/v1"
+	"github.com/openshift/origin/pkg/image/api"
+)
+
+func fuzzImage(t *testing.T, image *api.Image, seed int64) *api.Image {
+	f := apitesting.FuzzerFor(t, v1.SchemeGroupVersion, rand.NewSource(seed))
+	f.Funcs(
+		func(j *api.Image, c fuzz.Continue) {
+			c.FuzzNoCustom(j)
+			j.Annotations = make(map[string]string)
+			j.Labels = make(map[string]string)
+			j.Signatures = make([]api.ImageSignature, c.Rand.Intn(3)+2)
+			for i := range j.Signatures {
+				sign := &j.Signatures[i]
+				c.Fuzz(sign)
+			}
+			for i := 0; i < c.Rand.Intn(3)+2; i++ {
+				j.Labels[c.RandString()] = c.RandString()
+				j.Annotations[c.RandString()] = c.RandString()
+			}
+		},
+	)
+
+	updated := api.Image{}
+	f.Fuzz(&updated)
+	updated.Namespace = image.Namespace
+	updated.Name = image.Name
+
+	j, err := meta.TypeAccessor(image)
+	if err != nil {
+		t.Fatalf("Unexpected error %v for %#v", err, image)
+	}
+	j.SetKind("")
+	j.SetAPIVersion("")
+
+	return &updated
+}
+
+func TestSignatureStrategyPrepareForUpdate(t *testing.T) {
+	original := api.Image{
```

Comment:
@liggitt here's the reflective test. Is there anything else in `ObjectMeta` that needs to be preserved?